Reminder-App/Reminder-App/ReminderListViewController.swift

```diff
@@ -9,32 +9,217 @@ import UIKit
+                    self.refreshProgressView()
+                }
+            })
+        
         tableView.dataSource = reminderListDataSource
     }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        
+        let radius = view.bounds.size.width * 0.5 * 0.7
+        progressContainerView.layer.cornerRadius = radius
+        progressContainerView.layer.masksToBounds = true
+        self.refreshProgressView()
+        refreshBackground()
+    }
+    
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        
+        if let navigationController = navigationController,
+           navigationController.isToolbarHidden {
+            navigationController.setToolbarHidden(false, animated: animated)
+        }
+    }
+    
+    // MARK: Private Methods
+    private func addReminder() {
+        let storyboard = UIStoryboard(name: Self.mainStoryboardName, bundle: nil)
+        let detailViewController: ReminderDetailViewController = storyboard.instantiateViewController(identifier: Self.detailViewControllerIdentifier)
+        let reminder = Reminder(id: UUID().uuidString, title: "New Reminder", dueDate: Date())
+        
+        detailViewController.configure(with: reminder,
+                                       isNew: true,
+                                       addAction: {
+                                        reminder in
+                                        self.reminderListDataSource?.add(reminder, completion: {
+                                            (index) in
+                                            if let index = index {
+                                                DispatchQueue.main.async {
+                                                    self.tableView.insertRows(at: [IndexPath(row: index, section: 0)], with: .automatic)
+                                                    self.refreshProgressView()
+                                                }
+                                            }
+                                        })
+                                       })
+        
+        let navigationController = UINavigationController(rootViewController: detailViewController)
+        present(navigationController, animated: true, completion: nil)
+    }
+    
+    private func refreshProgressView() {
+        guard let percentComplete = reminderListDataSource?.percentComplete else {
+            return
+        }
+        
+        let totalHeight = progressContainerView.bounds.size.height
+        percentCompleteHeightConstraint.constant = totalHeight * CGFloat(percentComplete)
+        
+        UIView.animate(withDuration: 0.2) {
+            self.progressContainerView.layoutSubviews()
+        }
+    }
+    
+    private func refreshBackground() {
+        tableView.backgroundView = nil
+        let backgroundView = UIView()
+        tableView.backgroundView = backgroundView
+        
```

Comment:
remove extra line space

Reply:
Will discuss these "spacing" and "closure parameters names" placing in community meeting.